cmake/BLTMacros.cmake

```diff
@@ -760,27 +760,43 @@ macro(blt_append_custom_compiler_flag)
    if(NOT DEFINED arg_FLAGS_VAR)
       message( FATAL_ERROR "append_custom_compiler_flag macro requires FLAGS_VAR keyword and argument." )
    endif()
-
+   
    # Set the desired flags based on the compiler family   
-   if( DEFINED arg_CLANG AND COMPILER_FAMILY_IS_CLANG )
-      set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_CLANG} " )
-   elseif( DEFINED arg_XL AND COMPILER_FAMILY_IS_XL )
-      set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_XL} " )
-   elseif( DEFINED arg_INTEL AND COMPILER_FAMILY_IS_INTEL )
-      set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_INTEL} " )
-   elseif( DEFINED arg_GNU AND COMPILER_FAMILY_IS_GNU )
-      set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_GNU} " )
-   elseif( DEFINED arg_HCC AND COMPILER_FAMILY_IS_HCC )
-      set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_HCC} " )
-    elseif( DEFINED arg_PGI AND COMPILER_FAMILY_IS_PGI )
-      set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_PGI} " )
-   elseif( DEFINED arg_MSVC_INTEL AND COMPILER_FAMILY_IS_MSVC_INTEL )
+   # MSVC COMPILER FAMILY applies to C/C++ and Fortran
+   string(TOLOWER ${arg_FLAGS_VAR} lower_flag_var)
```

Comment:
Due to weirdness in cmake macros, please ``unset(lower_flag_var)`` at the bottom of the macro before returning.  Otherwise, this variable gets added to the caller's scope.

Reply:
Done.